haas/api.py

```diff
@@ -510,6 +510,23 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    """List all networks.
+    Returns a JSON array of strings representing a list of projects.
+    Example:  {"channels": ["101", "107"], "name": ["net-05", "qnet-00"]}
+    """
+    db = model.Session()
+    networks = db.query(model.Network).all()
+    result = {}
+    for n in networks:
+        if n.access is not None:
+            net = {'channel': n.network_id, 'project': n.access.label}
+        else:
+            net = {'channel': n.network_id, 'project': 'Public'}
```

Comment:
Sorry for the delay - we just had some discussion on this in #537. We decided to set project to the null entry (`None` in python) for objects without projects, so I think you should be able to use very similar logic here:
```python
'project': None if n.creator_id is None else n.creator.label
```